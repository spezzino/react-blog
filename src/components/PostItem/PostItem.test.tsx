import React from "react";
import { PostItem } from "./";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

test("PostItem renders", () => {
  const component = renderer.create(
    <PostItem
      id={1}
      title="Test title"
      body="Test body"
      user={{ id: 1, name: "User 1" }}
      showMoreButton={false}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("PostItem renders with View More Button", () => {
  const component = renderer.create(
    <MemoryRouter>
      <PostItem
        id={1}
        title="Test title"
        body="Test body"
        user={{ id: 1, name: "User 1" }}
        showMoreButton={true}
      />
    </MemoryRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(
    component.root.findByProps({ className: "MuiButton-label" }).children
  ).toEqual(["View more"]);
});
