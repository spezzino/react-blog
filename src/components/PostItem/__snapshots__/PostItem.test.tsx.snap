// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostItem renders 1`] = `
<div
  className="MuiPaper-root MuiCard-root list-item MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h5"
    >
      Test title
    </h2>
    <p
      className="MuiTypography-root pos MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      by 
      User 1
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2"
    >
      Test body
    </p>
  </div>
</div>
`;

exports[`PostItem renders with View More Button 1`] = `
<div
  className="MuiPaper-root MuiCard-root list-item MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h5"
    >
      Test title
    </h2>
    <p
      className="MuiTypography-root pos MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      by 
      User 1
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2"
    >
      Test body
    </p>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
      href="/posts/1"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiButton-label"
      >
        View more
      </span>
    </a>
  </div>
</div>
`;
